jwk.go

```diff
@@ -145,3 +166,56 @@ func (key *rawJsonWebKey) fromEcPublicKey(pub *ecdsa.PublicKey) error {
 
 	return nil
 }
+
+func (jwk rawJsonWebKey) rsaPrivateKey() (*rsa.PrivateKey, error) {
+	if jwk.D == nil || jwk.P == nil || jwk.Q == nil {
+		return nil, fmt.Errorf("square/go-jose: invalid RSA private key, missing values")
+	}
+
+	rv := new(rsa.PrivateKey)
+	pubkey, err := jwk.rsaPublicKey()
+	if err != nil {
+		return nil, err
+	}
+	rv.PublicKey = *pubkey
+	rv.D = jwk.D.bigInt()
+	rv.Primes = make([]*big.Int, 2)
+	rv.Primes[0] = jwk.P.bigInt()
+	rv.Primes[1] = jwk.Q.bigInt()
+
+	err = rv.Validate()
+	return rv, err
+}
+
+func (jwkp *rawJsonWebKey) fromRsaPrivateKey(rsa *rsa.PrivateKey) {
+	var jwk rawJsonWebKey
+	jwk.fromRsaPublicKey(&rsa.PublicKey)
+
+	jwk.D = newBuffer(rsa.D.Bytes())
+	// TODO: do we need to ensure len(Primes) >= 2 and/or deal with more than 2?
```

Comment:
As far as I can see, the JWK spec does not mention how to represent RSA keys with more than two primes. The library should probably check if a key has more than two primes and reject it.